packages/custom-gateway-bridging/contracts/L1CustomGateway.sol

```diff
@@ -0,0 +1,222 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.0;
+
+import "./interfaces/ICustomGateway.sol";
+import "./CrosschainMessenger.sol";
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+/**
+ * @title Example implementation of a custom gateway to be deployed on L1
+ * @dev Inheritance of Ownable is optional. In this case we use it to call the function setTokenBridgeInformation
+ * and simplify the test
+ */
+contract L1CustomGateway is IL1CustomGateway, L1CrosschainMessenger, Ownable {
+    
+    // Token bridge state variables
+    address public l1CustomToken;
+    address public l2CustomToken;
+    address public l2Gateway;
+    address public router;
+    bool private tokenBridgeInformationSet = false;
+
+    // Custom functionality
+    bool public allowsDeposits = false;
```

Comment:
while this is safe bcz false is the default value, setting value when defining storage var can be risky when you use a proxy as it won't be set

Reply:
This version is not prepared for a proxy, but in order to minimize the changes to be made to use a proxy, I've moved the default value to the constructor.